weave/flow/dataset.py

```diff
@@ -77,11 +78,11 @@ def to_pandas(self) -> "pd.DataFrame":
         return pd.DataFrame(self.rows)
 
     @field_validator("rows", mode="before")
-    def convert_to_table(cls, rows: Any) -> weave.Table:
+    def convert_to_table(cls, rows: Any) -> Union[weave.Table, WeaveTable]:
```

Comment:
nit: would be nice if this used casting:
https://github.com/wandb/weave/blob/master/weave/flow/casting.py

Then you can annotate anything to auto-convert to a table

Reply:
not exactly sure what this should look like, opened a draft for a fast follow: https://github.com/wandb/weave/pull/3796